packages/pluggableWidgets/switch-web/src/components/Switch.tsx

```diff
@@ -0,0 +1,47 @@
+import { MouseEvent, KeyboardEvent, createElement } from "react";
+import classNames from "classnames";
+import { Alert } from "@mendix/piw-utils-internal";
+import { SwitchContainerProps } from "../../typings/SwitchProps";
+
+export interface SwitchProps extends Pick<SwitchContainerProps, "id" | "class" | "style" | "tabIndex"> {
+    onClick?: (event: MouseEvent<HTMLDivElement>) => void;
+    onKeyDown?: (event: KeyboardEvent<HTMLDivElement>) => void;
+    isChecked: boolean;
+    editable: boolean;
+    validation: string | undefined;
+}
+
+export default function Switch(props: SwitchProps) {
+    return (
+        <div className={classNames("widget-switch", props.class)} style={props.style}>
+            <input
```

Comment:
Are you sure this element in combination with the div with class "widget-switch-btn-wrapper" with all its aria attributes is treated well in screen readers. It feels really duplicate, especially because in the switch example on MDN they use the button element: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Switch_role. I understand we want to keep DOM changes to a minimum, but maybe we could just get rid of this input altogether. I think we do apply styling to input elements in Atlas, so it could be difficult to get rid of it. I would suggest to atleast check with some screenreaders whether it's treated well.

Reply:
I removed `input`. After consulting with Jelmer, he agreed this has no purpose. It had `display: none` applied *always* so styling was not a concern.

Reply:
@Keraito if you can please review for Isa

Reply:
Addressed in https://github.com/mendix/widgets-resources/commit/63e36cb003bfec274e216591ff80160f1538a13f, @jordanch since I might be missing some context, do you know what the purpose of this `input` element was?

Reply:
Asking this because of the discussion we had several times regarding keeping the DOM structure the same before and after migration. Based on your comment it does seem useless and I'm all down for getting rid of it if it's not used in any meaningful way, but do want to verify that cc @diego-antonelli

Reply:
@Keraito to be honest I'm not sure _why_ it has been there for a long time. I don't know what purpose it served, nor does Jelmer. I thought maybe it had something to do with mxforms but not, the `EditableValue` in conjunction with the client widet framework handles values/validation. It seems to be legacy that has remained around.

Reply:
Perhaps it was used for accessibility in the past years.

> [Where possible use an HTML <input> element with type="checkbox" as this element has built in semantics and does not require ARIA attributes.](https://developer.mozilla.org/en-US/docs/Web/API/Element/ariaChecked) 

Then we introduced the `aria-` attributes, causing `input` to be redundant.

Reply:
Sounds reasonable to me 👍

Reply:
Great work 👍